### ISSUE TITLE
TransT code/没有TransT部分的代码吗

### ISSUE THREAD

**Issue by xiboliyalang-cc** [Kou-99/ZoomTrack#1]
Is the TransT part of the code not included?

**Comment by Kou-99** [comment]
> Is the TransT part of the code not included?

Yes. The code of TransT is not included since it uses a different repo and it would be too complicated to merge these 2 repos. You can modify the code of TransT following the OSTrack version of ZoomTrack. I recommend using the TransT-M version instead of the original TransT because TransT-M supports DDP which speeds up dataloading.

**Comment by Kou-99** [comment]
> @Kou-99 作者您好，十分感谢你的工作。 在阅读代码的过程中，发现ZoomTrack中unwarp_bboxes是对归一化的边界框操作的。请问如果在直接回归边界框的跟踪器上使用，unwarp_bboxes函数应该怎么修改呢？ 谢谢！

Sorry for the late reply. Maybe you can normalize the predicted bounding box on the resized image then use unwarp_bboxes to map it back to the original image. This could be a workaround to avoid modifying unwarp_bboxes.

**Comment by Kou-99** [comment]
> @Kou-99 Hello, author. Thanks for your kind help, I have solved the last problem. But a new problem arose. When I generated the grid with grid_shape=(17,17), the magnification was not obvious, so I raised grid_shape to (31, 3 1), the magnification effect is obvious, but an error will be reported after running for a while: `Rank(A) < p or Rank([P; A; G]) < n` May I ask what causes this? Thanks in advance!

Sorry for the late reply. I never met this error in my own experiments so I cannot give a direct reason or solution to this issue. The problem seems to relate to the qp solver / grid generator.  I recommend implementing try-except blocks to catch the error and print relevant matrices (P, q, A, b) along with the input to the grid generator.  This could help identify the root cause. I suspect that the error might be linked to the invalid input to the grid generator. Additionally, I suggest using the zoom factor (γ) to regulate the amount of magnification instead of relying on grid size since using the zoom factor provides more precise control over magnification levels.

**Comment by xyl-507** [comment]
@Kou-99 
Thanks to your careful guidance, the zoom factor (γ) can be used to accurately control the degree of amplification. 
But I have one more question, should I pay attention to adjusting any parameters when zooming in on **the whole image** (not the search area)? No matter how I adjust the zoom factor (γ), the magnification in the whole image doesn't work.
Looking forward to your reply!

**Comment by Kou-99** [comment]
> @Kou-99 Thanks to your careful guidance, the zoom factor (γ) can be used to accurately control the degree of amplification. But I have one more question, should I pay attention to adjusting any parameters when zooming in on **the whole image** (not the search area)? No matter how I adjust the zoom factor (γ), the magnification in the whole image doesn't work. Looking forward to your reply!

I did not do any experiments on the entire image. But I think there is a potential explanation for why the zoom factor doesn't work and why increasing grid size leads to better magnification. The magnification may be compromised during interpolation from the control grid to the dense grid when the size of the dense grid is notably greater than that of the control grid. So the grid size indeed should be adjusted accordingly if the target image is larger.

**Comment by xyl-507** [comment]
@Kou-99 
Thank you very much. 
I will try to solve the problem of enlarging the grid size.
